pkg/controller/openliberty/libertyapplication_controller.go

```diff
@@ -456,5 +456,35 @@ func (r *ReconcileOpenLiberty) Reconcile(request reconcile.Request) (reconcile.R
 		reqLogger.V(1).Info(fmt.Sprintf("%s is not supported", routev1.SchemeGroupVersion.String()))
 	}
 
+	if instance.Spec.CreateAppDefinition != nil && *instance.Spec.CreateAppDefinition {
+		m := make(map[string]string)
+		m["kappnav.subkind"] = "Liberty"
+		if instance.Spec.Storage != nil {
+			statefulSet := &appsv1.StatefulSet{ObjectMeta: defaultMeta}
```

Comment:
You can add this logic into existing place where we already define statefulset/deployment

Reply:
Added the changes into where statefulset and deployment are defined.